src/System.Memory/src/System/Buffers/Text/Base64.cs

```diff
@@ -0,0 +1,175 @@
+            {
+                s_avxEncodePermuteVec = Vector256.Create(0, 0, 1, 2, 3, 4, 5, 6);
+
+                s_avxEncodeShuffleVec = Vector256.Create(
+                     5, 4, 6, 5,
+                     8, 7, 9, 8,
+                    11, 10, 12, 11,
+                    14, 13, 15, 14,
+                     1, 0, 2, 1,
+                     4, 3, 5, 4,
+                     7, 6, 8, 7,
+                    10, 9, 11, 10
+                );
+
+                s_avxEncodeLut = Vector256.Create(
+                     65, 71, -4, -4,
+                     -4, -4, -4, -4,
+                     -4, -4, -4, -4,
+                    -19, -16, 0, 0,
+                     65, 71, -4, -4,
+                     -4, -4, -4, -4,
+                     -4, -4, -4, -4,
+                    -19, -16, 0, 0
+                );
+
+                s_avxDecodeLutLo = Vector256.Create(
+                    0x15, 0x11, 0x11, 0x11,
+                    0x11, 0x11, 0x11, 0x11,
+                    0x11, 0x11, 0x13, 0x1A,
+                    0x1B, 0x1B, 0x1B, 0x1A,
+                    0x15, 0x11, 0x11, 0x11,
+                    0x11, 0x11, 0x11, 0x11,
+                    0x11, 0x11, 0x13, 0x1A,
+                    0x1B, 0x1B, 0x1B, 0x1A
+                );
+
+                s_avxDecodeLutHi = Vector256.Create(
+                    0x10, 0x10, 0x01, 0x02,
+                    0x04, 0x08, 0x04, 0x08,
+                    0x10, 0x10, 0x10, 0x10,
+                    0x10, 0x10, 0x10, 0x10,
+                    0x10, 0x10, 0x01, 0x02,
+                    0x04, 0x08, 0x04, 0x08,
+                    0x10, 0x10, 0x10, 0x10,
+                    0x10, 0x10, 0x10, 0x10
+                );
+
+                s_avxDecodeLutShift = Vector256.Create(
+                     0, 16, 19, 4,
+                    -65, -65, -71, -71,
+                      0, 0, 0, 0,
+                      0, 0, 0, 0,
+                      0, 16, 19, 4,
+                    -65, -65, -71, -71,
+                      0, 0, 0, 0,
+                      0, 0, 0, 0
+                );
+
+                s_avxDecodeShuffleVec = Vector256.Create(
+                    2, 1, 0, 6,
+                    5, 4, 10, 9,
+                    8, 14, 13, 12,
+                   -1, -1, -1, -1,
+                    2, 1, 0, 6,
+                    5, 4, 10, 9,
+                    8, 14, 13, 12,
+                   -1, -1, -1, -1
+               );
+
+                s_avxDecodePermuteVec = Vector256.Create(0, 1, 2, 4, 5, 6, -1, -1);
+
+                s_avxDecodeMask2F = Vector256.Create((sbyte)0x2F);    // ASCII: /
+            }
+        }
+
+        [Conditional("DEBUG")]
+        private static void AssertRead<TVector>(ref byte src, ref byte srcStart, int srcLength)
+        {
+            int vectorElements = Unsafe.SizeOf<TVector>();
+            ref byte readEnd = ref Unsafe.Add(ref src, vectorElements);
+            ref byte srcEnd = ref Unsafe.Add(ref srcStart, srcLength + 1);
```

Comment:
The GC might not be able to track this ref. Since you're checking for error conditions, I recommend using raw pointers rather than GC-tracked refs, rewriting this method in terms of the 'fixed' statement. Also check for boundary conditions like integer overflows.

Reply:
(Same comment for AssertWrite.)